lib/authority_ecto/changeset.ex

```diff
@@ -2,4 +2,57 @@ defmodule Authority.Ecto.Changeset do
   @moduledoc """
   Convenient authentication-related functions for `Ecto.Changeset`s.
   """
-end
\ No newline at end of file
+
+  import Ecto.Changeset
+
+  @doc """
+  Hashes the value stored in the `source` field, and puts the resulting
+  hash in the `destination` field. The `source` field will be removed
+  from the changeset.
+
+  By default, the password will be hashed using `Comeonin.Bcrypt`. See
+  `put_encrypted_password/4` to use a different algorithm. Valid options
+  are `:bcrypt`, `:argon2`, or `:pbkdf2`.
+
+  ## Examples
+
+      iex> put_encrypted_password(changeset, :password, :encrypted_password)
+      %Ecto.Changeset{}
+
+      iex> put_encrypted_password(changeset, :password, :encrypted_password, :argon2)
+      %Ecto.Changeset{}
+
+  """
+  def put_encrypted_password(changeset, source, destination, algorithm \\ :bcrypt) do
+    password = get_change(changeset, source)
+    confirmation = String.to_atom(Atom.to_string(source) <> "_confirmation")
```

Comment:
A more secure version of this would be:

```elixir
confirmation = String.to_existing_atom("#{source}_confirmation")
```

`String.to_atom` is less safe than `String.to_existing_atom`, because it creates new atoms. If user input ever gets passed into `String.to_atom`, the code will have a denial-of-service vulnerability. Atoms aren't garbage collected, so if a user can create atoms, they can crash the server.

For this reason, it's probably safer to just never use `String.to_atom`.